t/selenium.js

```diff
@@ -270,50 +292,173 @@ async function handleCommand(command, target, value, baseURL, t) {
   }
 }
 
-const tests = [
-  'Create_Account.html',
-  'Log_Out.html',
-  'Log_In.html',
-  'MBS-7456.html',
-  'MBS-9548.html',
-  'Artist_Credit_Editor.html',
-  'External_Links_Editor.html',
-  'Work_Editor.html',
+const seleniumTests = [
+  {name: 'Create_Account.html'},
+  {name: 'MBS-7456.html', login: true},
+  {name: 'MBS-9548.html'},
+  {name: 'Artist_Credit_Editor.html', login: true},
+  {name: 'External_Links_Editor.html', login: true},
+  {name: 'Work_Editor.html', login: true},
 ];
 
-async function nextTest(testIndex) {
-  const file = path.resolve(__dirname, 'selenium', tests[testIndex]);
-  const {document} = new jsdom.JSDOM(fs.readFileSync(file)).window;
+const testPath = name => path.resolve(__dirname, 'selenium', name);
+
+seleniumTests.forEach(x => {
+  x.path = testPath(x.name);
+});
 
+function getPlan(file) {
+  const {document} = new jsdom.JSDOM(fs.readFileSync(file)).window;
   const baseURL = document.querySelector('link[rel=selenium\\.base]').href;
   const title = document.querySelector('title').textContent;
   const tbody = document.querySelector('tbody');
   const rows = Array.prototype.slice.call(tbody.getElementsByTagName('tr'), 0);
+  const commands = [];
+  let plan = 0;
 
-  test(title, {timeout: 60000}, async function (t) {
-    async function nextRow(index) {
-      if (index < rows.length) {
-        const cols = rows[index].getElementsByTagName('td');
-        const command = cols[0].textContent;
-        const target = cols[1].textContent;
-        const value = cols[2].textContent;
+  for (let i = 0; i < rows.length; i++) {
+    const cols = rows[i].getElementsByTagName('td');
+    const command = cols[0].textContent;
+    const target = cols[1].textContent;
+    const value = cols[2].textContent;
 
-        await handleCommand(command, target, value, baseURL, t);
+    if (/^assert/.test(command)) {
+      plan++;
+    }
 
-        return nextRow(index + 1);
-      } else {
-        t.end();
+    commands.push([command, target, value, baseURL]);
+  }
 
-        if (testIndex < tests.length - 1) {
-          process.nextTick(nextTest, testIndex + 1);
-        } else {
-          await quit();
-        }
-      }
-    }
+  return {commands, plan, title};
+}
 
-    await nextRow(0);
-  });
+async function runCommands(commands, t) {
+  for (let i = 0; i < commands.length; i++) {
+    await handleCommand(...commands[i], t);
+  }
 }
 
-nextTest(0);
+(async function runTests() {
+  const TEST_TIMEOUT = 60000; // 60 seconds
+
+  const cartonPrefix = process.env.PERL_CARTON_PATH
+    ? 'carton exec -- '
+    : '';
+
+  function pgPasswordEnv(db) {
+    if (db.password) {
+      return {env: {PGPASSWORD: db.password}};
```

Comment:
Shouldn’t it return `{PGPASSWORD: db.password}` only?

Reply:
I think not since it ultimately gets passed as the `options` argument to https://nodejs.org/api/child_process.html#child_process_child_process_execfile_file_args_options_callback

Reply:
Alright, returning broken options set misleadingly masked some errors for me. The actual issue is that some variables in the process environment are necessary to run shell commands on Linux. Cloning `process.env` allows to work around it.